PHP/PHP Source.sublime-syntax

```diff
@@ -683,7 +683,7 @@ contexts:
 ###[ CLASS DECLARATIONS ]#####################################################
 
   class-declarations:
-    - match: (?i:(abstract|final)\s+)?((?i:class))\b
+    - match: (?i:(abstract|final|readonly)\s+)?((?i:class))\b
```

Comment:
The easiest workaround was to remove modifiers from pattern at all. So `readonly` would be scoped correctly automatically and the declaration pattern would look exactly like those for enums and interfaces. The only downside is `meta.class` not covering those modifiers, which it currently does not for enums/interfaces/functions as well.

```yml
  class-declarations:
    - match: (?i:class)\b
      scope: meta.class.php keyword.declaration.class.php
      push:
        - class-declaration-body
        - class-declaration-name
```

Assuming modifiers are always written in the same line as the declaration keyword, an alternative (maybe for the others, too) could look like:

```yml
  class-declarations:
    - match: (?=(?:{{storage_modifiers}}\s+)*(?i:class)\b)
      push:
        - class-declaration-body
        - class-declaration-name
        - class-declaration-type

  class-declaration-type:
    - match: (?i:class)\b
      scope: keyword.declaration.class.php
      pop: 1
    - match: '{{storage_modifiers}}'
      scope: storage.modifier.php
```

In case modifiers may be on separate lines, we'd need an approach using `branch`es just like Java does.

Reply:
Thanks. I think the easy workaround is probably fine.